lib/galaxy/security/validate_user_input.py

```diff
@@ -45,7 +45,7 @@ def validate_publicname( trans, publicname, user=None ):
         if len( publicname ) < 3:
             return "Public name must be at least 3 characters in length"
     else:
-        if len( publicname ) < 4:
+        if len( publicname ) < 3:
             return "Public name must be at least 4 characters in length"
```

Comment:
Why is this changing? Also the condition does not correspond with the return message. 
Note that TS has 3 and Galaxy 4 char username min length limit.